From b83ae74b1d6ee1bb231e0680d9a68e470b23fb79 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Thu, 9 Apr 2026 00:12:09 +0100
Subject: [PATCH 2/2] sepolicy: ship plat_sepolicy_genfs_202404.cil for legacy
 vendor partitions

get_genfs_labels_version() defaults to 202404 when the vendor partition
does not have /vendor/etc/selinux/genfs_labels_version.txt. the build
currently only ships plat_sepolicy_genfs_202504.cil and _202604.cil,
so any device without the version file hits:

  init: Missing /system/etc/selinux/plat_sepolicy_genfs_202404.cil; skipping

this includes all older devices (Android 8-15 era) that have no
genfs_labels_version.txt in their vendor partition.

add an empty plat_sepolicy_genfs_202404.cil with the corresponding
prebuilt_etc build rule. empty is correct: the 202504 file was the
first to introduce versioned genfs overrides (sysfs_udc and
sysfs_mem_sleep); there were no new versioned genfs entries at the
202404 epoch.
---
 compat/Android.bp                     | 6 ++++++
 compat/plat_sepolicy_genfs_202404.cil | 0
 2 files changed, 6 insertions(+)
 create mode 100644 compat/plat_sepolicy_genfs_202404.cil

diff --git a/compat/Android.bp b/compat/Android.bp
index 4a144de..d68cd7b 100644
--- a/compat/Android.bp
+++ b/compat/Android.bp
@@ -660,6 +660,12 @@ se_compat_cil {
     version: "202404",
 }
 
+prebuilt_etc {
+    name: "plat_sepolicy_genfs_202404.cil",
+    src: "plat_sepolicy_genfs_202404.cil",
+    relative_install_path: "selinux",
+}
+
 prebuilt_etc {
     name: "plat_sepolicy_genfs_202504.cil",
     src: "plat_sepolicy_genfs_202504.cil",
diff --git a/compat/plat_sepolicy_genfs_202404.cil b/compat/plat_sepolicy_genfs_202404.cil
new file mode 100644
index 0000000..e69de29
-- 
2.53.0

